mantidproject/mantid
# Fix crash due to subscribing observer twice

**Report to:** Franz Demmel

### Description of work
This PR fixes a consistent crash when using multiple of the Indirect and Inelastic interfaces. It is possible to cause this crash in multiple different ways, from multiple different interfaces. It is caused by adding the same observer to the notification center more than once, and the solution was easy - not to add the observer if it is already subscribed to the notification center.

Fixes #37071


### To test:
1. Go to 'Inelastic'->`Data Manipulations`
2. Load data into the Symmetrise Tab
3. Click `Preview`, then click `Run`
4. Close the interface
5. Open the `Data Analysis` interface
6. On MSD Fit load data
7. Select `Gaus` function
8. Click `Run`. There should be no crash

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
